tier1/KeyValues.cpp

```diff
@@ -45,6 +45,30 @@ void KVStringDelete(void* pMem)
 	MemAlloc_Free(pMem);
 }
 
+static bool BKeyValuesSystemSupportsCache()
+{
+	static bool s_bSupportsCache = false;
+	static bool s_bCheckedForCacheSupport = false;
+
+	if (!s_bCheckedForCacheSupport)
+	{
+		// Use Sys_LoadModule to resolve actual bin name.
+		CSysModule *pTier0 = Sys_LoadModule("tier0");
+		if (pTier0)
+		{
+#ifdef _WIN32
+			s_bSupportsCache = !!GetProcAddress(reinterpret_cast<HMODULE>(pTier0), "HushAsserts");
+#elif defined(POSIX)
+			s_bSupportsCache = !!dlsym(reinterpret_cast<void *>(pTier0), "HushAsserts");
+#endif

```

Comment:
`Sys_GetProcAddress` ?

Reply:
The only Sys_GetProcAddress overload on Linux requires doing the module lookup again, so I just went native on all platforms.

See: https://github.com/alliedmodders/hl2sdk/blob/sdk2013/tier1/interface.cpp#L136